### PR TITLE
Preliminary bzlmod support

### LINKED ISSUE #63: Be able to depend on Gazelle Cabal via bzlmod.
This means that Gazelle Cabal should have a MODULE.bazel file that declares all the dependencies required by a user of Gazelle Cabal.

### PR DESCRIPTION
**Pull request by aherrmann**
This configures the `gazelle_cabal` workspace to be compatible with bzlmod as a dependency module. Meaning another module can depend on it via bzlmod, however `gazelle_cabal` cannot itself be tested or built with bzlmod enabled, yet.

This also configures the `example` workspace to be compatible with bzlmod as a root module. Meaning the example can be built and tested with bzlmod enabled.

Enables bzlmod tests on CI, for now only for GHC version 9.4.5.

**limitations / future work**
- The root module cannot itself be tested or built with bzlmod, yet.
- The `tests` are not built with bzlmod, yet.
    The `alternative-deps` test requires renaming the stack snapshot workspace names to `stackage-b` and `stackage-b-exe`. This is currently not possible. One path would be a `name` attribute to `stack_snapshot`. However, this is not desirable as discussed in https://github.com/tweag/rules_haskell/pull/1962. Another approach is to use isolated extensions (requires Bazel 6.3) and remapping on `use_repo`. But, `use_repo` uses keyword argument syntax for the remapping target, and `stackage-b` or `stackage-b-exe` do (AFAIK) not form valid identifiers in Starlark.
- The GHC version is hard-coded in the MODULE.bazel file and fixed to 9.4.5.
    The version number is factored out into a variable. A simple future improvement would be to simply change that variable in place in the CI pipeline to test other versions.
- The Gazelle extension does not itself support bzlmod, yet. In particular, `update-repos` does assume a `WORKSPACE` setup.

closes #63 #66

### CODE REVIEW

**Comment by aherrmann** [comment]
This will require an update of the required CI checks on GitHub.

File: MODULE.bazel [hunk 1]
```diff
@@ -5,14 +5,14 @@ module(
...
     "stack_snapshot",
 )
-use_repo(stack_snapshots, "stackage")
-stack_snapshots.package(name = "directory")
-stack_snapshots.package(name = "filepath")
-stack_snapshots.package(name = "json")
+use_repo(stack_snapshot, "stackage")
+stack_snapshot.package(name = "directory")
+stack_snapshot.package(name = "filepath")
+stack_snapshot.package(name = "json")
```
Comment (avdv) [inline MODULE.bazel:15]: That's nicer :+1:

**Review by avdv — APPROVED** [review]
Thank you!

**Comment by aherrmann** [comment]
@Mergifyio rebase

**Pull request merged**